### ISSUE TITLE
NEOS interface broken for certain solvers 

### ISSUE THREAD

**Issue by mlangiu** [Pyomo/pyomo#910]
Hey I just noticed that the NEOS interface seems to be broken, here's a minimal working example:

```python
from pyomo.environ import *
m = ConcreteModel()
m.x = Var()
m.y = Var()
m.c = ConstraintList()
m.c.add(2 * m.x >= m.y)
m.o = Objective(expr=m.x**2 + m.y**2)


def solve(solver):
    with SolverManagerFactory('neos') as manager:
        opt = SolverFactory(solver, validate=False)
        results = manager.solve(m, opt=opt)


solve('minlp')  # ERROR: Error parsing NEOS solution file
solve('knitro')  # Error parsing NEOS solution file
solve('snopt')  # ERROR: Error loading NEOS solution into model
```

My Versions:
OS: Win10 Enterprise
```python
print(pyomo.version.version)  # 5.5.1
import sys
print(sys.version)
# '3.6.7 |Anaconda, Inc.| (default, Dec 10 2018, 20:35:02) [MSC v.1915 64 bit (AMD64)]'
```

**Comment by qtothec** [comment]
I don't personally use NEOS, so I'll wait to have someone else take a look. In the meantime, you could see if it's resolvable by upgrading to Pyomo 5.6.1.

**Comment by mlangiu** [comment]
I just tried with 5.6.1 on both Mac and Windows, but no luck

**Comment by blnicho** [comment]
According to @jsiirola there does appear to be an issue with the NEOS interface with the three solvers mentioned above but all of the other NEOS solvers should work fine. We don't have access to these solvers outside of NEOS to easily debug this.

**Comment by ja-b** [comment]
I'm seeing this behavior for every solver right now. Not sure if its related to a temporary issue with NEOS.

**Comment by ja-b** [comment]
I'm seeing the same behavior for other various MINLP solvers, Bonmin, Couenne etc.

Specifically, it looks like NEOS returns a zip file without any contents within it, no sol file.

**Comment by blnicho** [comment]
@ja-b could you try re-running your model with the solvers you mentioned? We regularly see intermittent NEOS server outages in our automated testing infrastructure and I'm wondering if that is what you encountered.

**Comment by ja-b** [comment]
@blnicho Yes, I’ve been noticing the outages as well. I have switched to passing a string parameter (eg. “bonmin”) into opt and has since been working fine. I was passing in the solver instance itself, much like OP, which was giving me troubles.

**Comment by salvapineda** [comment]
I had the same experience as @ja-b. If I try to solve model instance m doing

res = SolverManagerFactory('neos').solve(m,opt=SolverFactory('knitro'))

I get an error. However, if I do

res = SolverManagerFactory('neos').solve(m,opt='knitro')

The optimization problem is solved correctly.

**Comment by djgray101** [comment]
Has anyone else still found this problem to be present? I have been trying to access the MINLP solvers but unfortunately none of them seem to be working for me.

**Comment by blnicho** [comment]
I think this has now been fixed by #1802
